src/arns.lua

```diff
@@ -839,130 +874,59 @@ function arns.assertValidIncreaseUndername(record, qty, currentTimestamp)
 	assert(qty > 0 and utils.isInteger(qty), "Qty is invalid")
 end
 
---- Creates an auction for a given name
---- @param name string The name of the auction
---- @param timestamp number The timestamp to start the auction
---- @param initiator string The address of the initiator of the auction
---- @return Auction|nil auction - the auction instance
-function arns.createAuction(name, timestamp, initiator)
-	assert(not arns.getRecord(name), "Name is registered. Auctions can only be created for unregistered names.")
-	assert(not arns.getReservedName(name), "Name is reserved. Auctions can only be created for unregistered names.")
-	assert(not arns.getAuction(name), "Auction already exists for name")
-	local baseFee = demand.baseFeeForNameLength(#name)
-	local demandFactor = demand.getDemandFactor()
-	local auction = Auction:new(name, timestamp, demandFactor, baseFee, initiator, arns.calculateRegistrationFee)
-	NameRegistry.auctions[name] = auction
-	arns.scheduleNextAuctionsPrune(auction.endTimestamp)
-	return auction
+--- Adds name to the recently returned name list
+--- @param name string The name of the returned name
+--- @param timestamp number The timestamp of the release
+--- @param initiator string The address of the initiator
+--- @returns ReturnedName
+function arns.createReturnedName(name, timestamp, initiator)
+	assert(not arns.getRecord(name), "Name is registered. Returned names can only be created for unregistered names.")
+	assert(
+		not arns.getReservedName(name),
+		"Name is reserved. Returned names can only be created for unregistered names."
+	)
+	assert(not arns.getReturnedName(name), "Returned name already exists")
+	local returnedName = {
+		name = name,
+		startTimestamp = timestamp,
+		initiator = initiator,
+	}
+	NameRegistry.returned[name] = returnedName
+	arns.scheduleNextReturnedNamesPrune(timestamp + constants.returnedNamePeriod)
+	return returnedName
 end
 
---- Gets an auction by name
---- @param name string The name of the auction
---- @return Auction|nil auction - the auction instance
-function arns.getAuction(name)
-	return NameRegistry.auctions[name]
+--- Gets a returned name
+--- @param name string The name of the returned name
+--- @return ReturnedName|nil
+function arns.getReturnedName(name)
```

Comment:
Consider adding "Unsafe" to the name here since it returns the actual table and not a copy. OR return a copy.

Reply:
2b61757